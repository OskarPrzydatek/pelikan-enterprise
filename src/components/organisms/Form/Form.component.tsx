import { Header } from '~/components/atoms';
import { IComponent } from '~/models';

import * as S from './Form.styles';

interface IForm extends IComponent {
  title: string;
  submitLabel: string;
  children: React.ReactNode;
  onSubmit: () => void;
}

export const Form: React.FC<IForm> = ({
  title,
  submitLabel,
  children,
  onSubmit,
}: IForm) => {
  return (
    <S.Form onSubmit={onSubmit}>
      <Header>{title}</Header>
      {children}
      <S.SubmitButton type="submit" value={submitLabel} />
    </S.Form>
  );
};
