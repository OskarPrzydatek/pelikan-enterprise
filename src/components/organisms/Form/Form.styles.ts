import styled from 'styled-components';

export const Form = styled.form`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: ${({ theme }) => theme.spaces.m};

  @media ${({ theme }) => theme.breakpoints.tablet} {
    width: 50%;
  }
`;

export const SubmitButton = styled.input`
  appearance: none;
  border: none;
  width: 100%;
  cursor: pointer;
  font-family: ${({ theme }) => theme.fonts.main};
  color: ${({ theme }) => theme.colors.white};
  background-color: ${({ theme }) => theme.colors.darkBlue};
  padding: ${({ theme }) => theme.spaces.s};
  font-size: ${({ theme }) => theme.fontSizes.s};

  @media ${({ theme }) => theme.breakpoints.tablet} {
    font-size: ${({ theme }) => theme.fontSizes.m};
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    width: 50%;
  }
`;
