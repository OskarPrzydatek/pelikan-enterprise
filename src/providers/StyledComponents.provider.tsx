import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
