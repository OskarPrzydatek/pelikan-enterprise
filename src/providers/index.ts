export { StyledComponentsProvider } from './StyledComponents.provider';
