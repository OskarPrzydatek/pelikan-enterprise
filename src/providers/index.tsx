import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
