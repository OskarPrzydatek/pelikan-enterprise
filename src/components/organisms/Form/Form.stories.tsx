import { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { useForm } from 'react-hook-form';

import { TextField } from '~/components/molecules';
import { StyledComponentsProvider } from '~/providers';

import { Form } from './Form';

const meta: Meta<typeof Form> = {
  title: 'Organisms/Form',
  component: Form,
};

export default meta;
type Story = StoryObj<typeof Form>;

const FormHelper: React.FC<{ isError?: boolean }> = ({ isError }) => {
  const methods = useForm<{ name: string }>();

  React.useEffect(() => {
    if (isError)
      methods.setError('name', {
        type: 'required',
        message: 'This is required',
      });
  }, [isError, methods]);

  return (
    <StyledComponentsProvider>
      <Form
        methods={methods}
        submitLabel="Submit Label"
        title="Form Helper"
        onSubmit={() => console.log('on submit')}
      >
        <TextField label="Name" name="name" registerOptions={{}} />
      </Form>
    </StyledComponentsProvider>
  );
};

export const FormStory: Story = {
  render: () => <FormHelper />,
};
