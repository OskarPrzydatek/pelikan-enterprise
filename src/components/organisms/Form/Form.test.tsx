import { render } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';

import { StyledComponentsProvider } from '~/providers';

import { Form } from './Form';

const mockOnSubmit = vi.fn();

const MockForm = () => (
  <StyledComponentsProvider>
    <Form
      methods={undefined}
      submitLabel="Submit label"
      title="Form title"
      onSubmit={mockOnSubmit}
    >
      <input />
    </Form>
  </StyledComponentsProvider>
);

describe('Form', () => {
  test('component snapshot', () => {
    const view = render(<MockForm />);
    expect(view).toMatchSnapshot();
  });
});
