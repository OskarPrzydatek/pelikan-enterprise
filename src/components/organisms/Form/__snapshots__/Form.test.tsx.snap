// Vitest Snapshot v1

exports[`Form > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-pyfCe rsgVC"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Form title
        </h2>
        <input />
        <input
          class="sc-jrcTuL iZHkcL"
          type="submit"
          value="Submit label"
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-pyfCe rsgVC"
    >
      <h2
        class="sc-eDvSVe dcXtK"
      >
        Form title
      </h2>
      <input />
      <input
        class="sc-jrcTuL iZHkcL"
        type="submit"
        value="Submit label"
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
